cpp/src/replace/replace.cu

```diff
@@ -13,26 +16,551 @@
+        }
+      }else{
+        if (replacement_has_nulls){
+          replace = replace_kernel<col_type, false, true>;
+        }else{
+          replace = replace_kernel<col_type, false, false>;
+        }
+      }
+      replace<<<grid.num_blocks, BLOCK_SIZE, 0, stream>>>(
+                                             input_col.data<col_type>(),
+                                             input_col.null_mask(),
+                                             output.data<col_type>(),
+                                             output.null_mask(),
+                                             valid_count,
+                                             output.size(),
+                                             values_to_replace.data<col_type>(),
+                                             values_to_replace.data<col_type>() + replacement_values.size(),
+                                             replacement_values.data<col_type>(),
+                                             replacement_values.null_mask());
+
+      cudaStreamSynchronize(stream);
+
+      if(valid_count != nullptr){
+        cudf::size_type valids {0};
+        CUDA_TRY(cudaMemcpyAsync(&valids,
+                                 valid_count,
+                                 sizeof(cudf::size_type),
+                                 cudaMemcpyDefault,
+                                 stream));
+        output.set_null_count(output.size() - valids);
+        mr->deallocate(valid_count, sizeof(cudf::size_type), stream);
+      }
+    }
+  };
+
+  template<>
+  void replace_kernel_forwarder::operator()<cudf::string_view> (cudf::column_view const& input_col,
+                                                                cudf::column_view const& values_to_replace,
+                                                                cudf::column_view const& replacement_values,
+                                                                cudf::mutable_column_view& output,
+                                                                cudaStream_t stream,
+                                                                rmm::mr::device_memory_resource* mr) {
+    CUDF_FAIL("Strings are not supported yet for replacement.");
+  }
+
+ } //end anonymous namespace
+
+namespace cudf{
+namespace detail {
+  std::unique_ptr<cudf::column> find_and_replace_all(cudf::column_view const& input_col,
+                                                     cudf::column_view const& values_to_replace,
+                                                     cudf::column_view const& replacement_values,
+                                                     rmm::mr::device_memory_resource* mr,
+                                                     cudaStream_t stream) {
+    if (0 == input_col.size() )
+    {
+      return std::make_unique<cudf::column>(input_col);
+    }
+
+    if (0 == values_to_replace.size() || 0 == replacement_values.size())
```

Comment:
You should first verify that `values_to_replace.size() == replacement_value.size()`

Reply:
Done.